frontend/app/components/Search/SearchBar/download-csv.tsx

```diff
@@ -0,0 +1,34 @@
+import { ArrowDownTrayIcon } from "@heroicons/react/24/outline";
+import { useHits } from "react-instantsearch";
+import extractAdDisclosure from "./extract-ad-disclosure";
+import jsonToCsv from "utils/json-to-csv";
+
+const DownloadCsv = () => {
+  const { hits } = useHits();
+
+  const handleClick = () => {
+    const csv = jsonToCsv(hits.map((hit) => extractAdDisclosure(hit)));
+    const blob = new Blob([csv], { type: "text/csv;charset=utf-8;" });
+    const link = document.createElement("a");
+    const url = URL.createObjectURL(blob);
+    link.setAttribute("href", url);
```

Comment:
Ah, so this is being handled completely on the front end? If this works, I'm good with it. 😁 

One non-blocking suggestion: perhaps you could extract the page link handling to another function so your `handleClick` wouldn't have so much logic in it.


```js
const handleClick = () => {
  const csv = jsonToCsv(...);
  const blob = makeBlob(...);
  const createDownloadLink(...);
  const downloadCsv(...);
}
```

It seems a bit much, but it would make it more clear what's happening, and each of those functions is more testable (not a concern now).